includes/class-smaily-for-wp-lifecycle.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * Define all the logic related to plugin activation, upgrade and uninstall logic.
+ *
+ * @since      3.0.0
+ * @package    Smaily_For_WP
+ * @subpackage Smaily_For_WP/includes
+ */
+class Smaily_For_WP_Lifecycle {
+
+	/**
+	 * Callback for plugin activation hook.
+	 *
+	 * @since 3.0.0
+	 */
+	public function activate() {
+		$this->run_migrations();
+
+		global $wpdb;
+		require_once ABSPATH . 'wp-admin/includes/upgrade.php';
+		$charset_collate = $wpdb->get_charset_collate();
+
+		// Create database table - settings.
+		$table_name = esc_sql( $wpdb->prefix . 'smaily_config' );
+		$sql        = "CREATE TABLE $table_name (
+			api_credentials VARCHAR(128) NOT NULL,
+			domain VARCHAR(255) NOT NULL,
+			form TEXT NOT NULL,
+			is_advanced TINYINT(1) NOT NULL,
+			PRIMARY KEY  (api_credentials)
+		) $charset_collate;";
+		dbDelta( $sql );
+	}
+
+	/**
+	 * Callback for plugins_loaded hook.
+	 *
+	 * Start migrations if plugin was updated.
+	 *
+	 * @since 3.0.0
+	 */
+	public function update() {
+		if ( get_transient( 'smailyforwp_plugin_updated' ) !== true ) {
+			return;
+		}
+		$this->run_migrations();
+		delete_transient( 'smailyforwp_plugin_updated' );
+	}
+
+	/**
+	 * Callback for plugin uninstall hook.
+	 *
+	 * Clean up plugin's database entities.
+	 *
+	 * @since 3.0.0
+	 */
+	public static function uninstall() {
+		global $wpdb;
+		$wpdb->query( "DROP TABLE IF EXISTS {$wpdb->prefix}smaily_config" );
+		$wpdb->query( "DROP TABLE IF EXISTS {$wpdb->prefix}smaily_autoresponder" );
```

Comment:
`wp_smaily_autoresponder` shouldn't exist for version 3.0.0, because 3.0.0 migration drops it and there shouldn't be a case where 3.0.0 codebase is installed, but migration has not been applied. You can safely omit dropping `wp_smaily_autoresponder` table here.

Reply:
That's right, dropped it in 1e5bec6